### DISCUSSION TITLE
Access possible only per MODBUS or also per REST-Server (HtREST)?

### DISCUSSION THREAD

**Discussion by cmanblues** [mbuchber/ha_heliotherm#5] [General]
Hi!
I have a Heliotherm HP08S10W-WEB heat pump and I am trying to implement it into Home Assistant somehow. 
I have no Heliotherm Remote Control unit or so installed in my heat pump. I used "HtREST" from dstrigl to connect to my heat pump (https://github.com/dstrigl/HtREST) via RS232.
I have a Raspberry PI which is connected with a Nullmodem-Cable to the RS232 port of the heat pump. And on the Raspberry Pi there is HtREST running, which work very well to get various parameters from the heat pump.

With your ha_heliotherm solution, is it somehow possible to work with the heat pump over this REST Server or is it only possible with MODBUS? As nice as HtREST is, I need a Home Assistant Integration to work with it...

Thx for your answer,
Christoph

**Comment by mbuchber** [comment]
Hi, sorry, I only implemented MODBUS protocol, as I only have this one working in my home. I don't have the REST Api running, thus cannot write any adapter to it - also if I wanted to.
Best regards,
 Michael

**Reply by cmanblues** [reply]
Thx for your quick answer.
You write that I need either an RS232-MODBUS Gateway or the RCG-Interface for your integration to work.
- If I buy this RS232 Gateway do I connect it then directly to the RS232 port of the heat pump? 
- How do I configure the Gateway so that it outputs correctly over MODBUS that your integration gets the right data? Because withou the RCG-Interface there is no Webserver of the heat pump to log in to and make some settings.
- I am also interested to see how many watts and kWh the heat pump uses / has. This info is available in some data points of the heat pump. Is this data available in your integration, or dont you read all data points from thw heat pump?

Thank you in advance for your feedback!
Christoph

**Comment by mbuchber** [comment]
- If I buy this RS232 Gateway do I connect it then directly to the RS232 port of the heat pump?
  Yes.

- How do I configure the Gateway so that it outputs correctly over MODBUS that your integration gets the right data? Because withou the RCG-Interface there is no Webserver of the heat pump to log in to and make some settings.
  Please read: https://github.com/mbuchber/ha_heliotherm#activating-modbus-tcp-using-heliotherm-webinterface

- I am also interested to see how many watts and kWh the heat pump uses / has. This info is available in some data points of the heat pump. Is this data available in your integration, or dont you read all data points from thw heat pump?
  Yes, all datapoint are read, also kWh - at least some users reported that it works (not warranty however!).

**Reply by mbuchber** [reply]
You only need to activate ModBus in the Webinterface of the RCG - a direct RS232 <-> ModBus interface does not need to be activated - it's the only "duty" of such interfaces ;) However, in order to avoid conflicts, I would always go for the "original" RCG interface, it is more expensive though.

**Reply by Johndowne** [reply]
I’m getting the kWh electricity consumed and the kWh heat generated through this integration and as far as I tell they look ok. I get around 4.2 CoP at the moment. The only issue with the numbers is that only complete kWhs are recorded so that if on one day it uses, say, 5.8kWh of electricity you just get the 5. The next day when the kWh is fully used, the figure increases to 6.
